Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift

```diff
@@ -120,20 +116,28 @@ class HomeViewController: UIViewController {
         else{
             NSLog("last player state wasn't updated properly, is nil")
         }
-        // get URI from algorithm, which is hard coded for now
+        // get URI from algorithm
         let alg_instance = songAlgorithm()
-        let songURI = alg_instance.getRandomSong()
-        // convert that to Spotify Song Object
-        _ = self.api_instance.appRemote.contentAPI?.fetchContentItem(forURI: songURI, callback: {success, error in
-            
-            if (error != nil) {
-                NSLog(error?.localizedDescription ?? "error fetching song")
+        alg_instance.getRandomSong {uri, error in
+            if error == nil, let uri = uri as? String?{
+                let songURI = uri
+                NSLog(uri ?? "nil uri call")
+                self.api_instance.appRemote.contentAPI?.fetchContentItem(forURI: songURI!, callback: {success, error in
+
+                    if (error != nil) {
+                        NSLog(error?.localizedDescription ?? "error fetching song")
+                    }
+                    else if let success = success as? SPTAppRemoteContentItem
+                    {
+                        //play if no errors
+                        self.api_instance.appRemote.playerAPI?.play(success)                    }
```

Comment:
Nit: playerAPI.play(success) 
does not read well, maybe consider changing to some meaningful name like 'song' or 'songContent' etc